app/preferences_ios.go

```diff
@@ -0,0 +1,126 @@
+// +build ios
+
+package app
+
+import (
+	"unsafe"
+
+	"fyne.io/fyne"
+)
+
+/*
+#import <stdlib.h>
+#import <Foundation/Foundation.h>
+
+int getPreferenceBool(const char* key, int fallback);
+void setPreferenceBool(const char* key, int value);
+float getPreferenceFloat(const char* key, float fallback);
+void setPreferenceFloat(const char* key, float value);
+int getPreferenceInt(const char* key, int fallback);
+void setPreferenceInt(const char* key, int value);
+const char* getPreferenceString(const char* key, const char* fallback);
+void setPreferenceString(const char* key, const char* value);
+*/
+import "C"
+
+type iOSPreferences struct {
+}
+
+// Declare conformity with Preferences interface
+var _ fyne.Preferences = (*iOSPreferences)(nil)
+
+func (p *iOSPreferences) Bool(key string) bool {
+	return p.BoolWithFallback(key, false)
+}
+
+func (p *iOSPreferences) BoolWithFallback(key string, fallback bool) bool {
+	cKey := C.CString(key)
+	cBool := 0
+	if fallback {
+		cBool = 1
+	}
+	ret := C.getPreferenceBool(cKey, C.int(cBool))
+
+	C.free(unsafe.Pointer(cKey))
```

Comment:
place the `free` near the `alloc`by using `defer` (applies to all following cKeys, too).
```suggestion
	cKey := C.CString(key)
	defer C.free(unsafe.Pointer(cKey))
	cBool := 0
	if fallback {
		cBool = 1
	}
	ret := C.getPreferenceBool(cKey, C.int(cBool))
```